app/controllers/foreman_leapp/api/v2/report_controller.rb

```diff
@@ -0,0 +1,25 @@
+module ForemanLeapp
+  module Api
+    module V2
+      class ReportController < ::Api::V2::BaseController
+  
+        def create
+          require 'date'
+          date = DateTime.now
+          host_name_or_id = params['host']
+          # XXX find appropriate host by id or name
+          host = Host.find_by(name: host_name_or_id) || Host.find_by(id: host_name_or_id)
```

Comment:
This could be done by a single query if you don't do not explicitly need that "first name then id" ordering `Host.where(:name => host_name_or_id).or(Host.where(:id => host_name_or_id)).first`